lib/services/all_question_service.dart

```diff
@@ -1,50 +1,38 @@
+import 'dart:async';
+import 'dart:convert';
+import 'package:http/http.dart' as http;
 
 class AllQuestionService {
-  // Here Eureka will get data from the database.
-  // At this point it contains just dummy lists of questions for test purposes
-  Future<List> getQuestions() async {
-    List data = [];
-    // List of 2 dummy questions
-    List list1 = List.generate(2, (index) => "Message")
-        .map((val) => QuestionModel(
-              category: "Technology",
-              date: DateTime.now(),
-              status: 1,
-              visible: 1,
-              title: "Linux installation issue - usb not found",
-              description:
-                  "Hello. I have a Smartbuy 16Gb USB 2.0 flash drive (with new memory controller) that is not recognized in any Linux system, but on Windows it recognized and worked fine. When I connect it to the PC on Linux system, nothing happens",
-            ))
-        .toList();
 
-    // List of another 2 dummy questions (different)
-    List list2 = List.generate(2, (index) => "Message")
-        .map((val) => QuestionModel(
-              category: "Household",
-              date: DateTime.now(),
-              status: 0,
-              visible: 1,
-              title: "Windows Issue",
-              description: "Hello",
-            ))
-        .toList();
+  // GET
+  Future<List<QuestionModel>> getQuestions() async {
 
-    // List of another 2 dummy questions (different)
-    List list3 = List.generate(2, (index) => "Message")
-        .map((val) => QuestionModel(
-              category: "Technology",
-              date: DateTime.now(),
-              status: 1,
-              visible: 1,
-              title: "iPhone not working",
-              description: "iphone stopped working",
-            ))
-        .toList();
+    await DotEnv.load();
 
-    // Combined 6 dummy questions together
-    List list2list3 = new List.from(list2)..addAll(list3);
-    data = new List.from(list1)..addAll(list2list3);
-    return data;
+    final response = await http.get(Uri.http(DotEnv.env['HOST'] + ':' + DotEnv.env['PORT'], '/v1/questions'));
+
+    if (response.statusCode == 200) {
+      // If the server did return a 200 OK response, then parse the JSON.
+      // Backend service returns a list of json instances, while we need a list of question objects.
+      // JSON parser will go through each json instance, transform it to a question object,
+      //   and append to the list that will be used in eureka_list_view
+      // Loop is reversed, as we want to sort the questions by date and show the most recent questions on top
+      //   (new questions are appending to the bottom of the database)
+
+      final body = json.decode(response.body);
+      print("All questions were loaded");
+
+      List<QuestionModel> questions = new List();
+      for (var i = body.length - 1; i >= 0; i--) {
+        questions.add(QuestionModel.fromJson(body[i]));
+      }
```

Comment:
I understand that the questions are sorted by active status. Recently posted questions are at the top. However, the two questions under my question would be out of order based of the time they were asked. If the a question was posted 21 hours ago why is it under a question that is 12 days old. Both have active status. 

![image](https://user-images.githubusercontent.com/38572314/112236424-a66bc700-8bfd-11eb-8e74-5dc0a8209a84.png)

Reply:
New questions are always going to have the most recent time. Flutter does not sort by date, but it is obvious that the last question in the list has the most recent timestamp. Since we are testing uploading questions in Postman, timestamp sometimes is not changed to the current time so that's why you see 12 days ago being above 21 hours ago, while that would never happen.

Reply:
Sounds good. If you're going to grab the current time and date within flutter. It would make it easier to make a POST request for the user. Removing this confusion of miss ordered questions.

Reply:
That is a separate task from this

Reply:
> Sounds good. If you're going to grab the current time and date within flutter. It would make it easier to make a POST request for the user. Removing this confusion of miss ordered questions.

Current time in the New Question Form that is being in development is taking the current time, so user won't interact with it. There won't be any confusions as everything is automated